runtime/chxvm_var.cc

```diff
@@ -41,9 +43,33 @@ ChxVMVar::ChxVMVar(chainerx::Shape shape) : kind_(Kind::kShape), val_(shape) {
 }
 
 const chainerx::Array& ChxVMVar::GetArray() const {
-    if (kind_ == Kind::kShape) {
-        kind_ = Kind::kArray;
-        val_ = runtime::ShapeToArray(absl::get<chainerx::Shape>(val_));
+    switch (kind_) {
+        case Kind::kShape:
+            kind_ = Kind::kArray;
+            val_ = runtime::ShapeToArray(absl::get<chainerx::Shape>(val_));
+            break;
+        case Kind::kScalar: {
+            const chainerx::Scalar& s = absl::get<chainerx::Scalar>(val_);
+            switch (s.kind()) {
+                case chainerx::DtypeKind::kBool: {
+                    bool b(s);
+                    val_ = runtime::MakeArray(chainerx::Dtype::kBool, {}, &b);
+                    break;
+                }
+                case chainerx::DtypeKind::kInt: {
+                    int64_t i(s);
+                    val_ = runtime::MakeArray(chainerx::Dtype::kInt64, {}, &i);
+                    break;
+                }
+                case chainerx::DtypeKind::kFloat: {
+                    double f(s);
+                    val_ = runtime::MakeArray(chainerx::Dtype::kFloat64, {}, &f);
```

Comment:
I think this can cause unexpected dtype promotions. Maybe scalar type in `VarInternalType` must be a pair of `chainerx::Scalar` and `chainerx::Dtype`?

Reply:
Yes. And in many cases it affects the type inferences too.
So I'm defining new type to represent scalar type.